File: Backend/src/main/java/com/example/demo/DTO/UploadDTO.java
package com.example.demo.DTO;

import com.example.demo.Models.Libro;
import com.example.demo.Models.Usuario;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
@Data
@NoArgsConstructor
@AllArgsConstructor

public class UploadDTO {
    @Id
    private Integer id;
    private Date fechaCreacion;
    private Integer rate;
    private String descripcion;
    private String titulo;
    //
    private Libro l;
    private UsuarioDTO u;
}

File: Frontend/src/app/Models/UploadDTO.ts
import { Libro } from "./Libro"
import { Usuario } from "./Usuario"

export class UploadDTO{
    id!: number
    fechaCreacion!: Date
    rate!: number 
    descripcion!: string 
    titulo!: string 
    u!: Usuario
    l!: Libro
}